sql/core/src/main/scala/org/apache/spark/sql/UDFRegistration.scala

```diff
@@ -86,9 +104,9 @@ class UDFRegistration private[sql] (sqlContext: SQLContext) extends Logging {
         def register[$typeTags](name: String, func: Function$x[$types]): UserDefinedFunction = {
           val dataType = ScalaReflection.schemaFor[RT].dataType
           val inputTypes = Try($inputTypes).getOrElse(Nil)
-          def builder(e: Seq[Expression]) = ScalaUDF(func, dataType, e, inputTypes)
```

Comment:
@marmbrus This kind of `builder`s caused the mima failure. Now, I am adding a build function in `ScalaUDF` because we need to get the value of `nondeterministic`.